### PR TITLE
Added deny-all default network policy

### PR DESCRIPTION
**Pull request by nsilla**
Done:
- Set default deny-all policy

To do:
- Identify and set allow policies. (For instance, cnf-app-mac-operator requires opening port 8095/TCP).

### CODE REVIEW

**Review by manurodriguez — COMMENTED** [review]
I checked the last job, but looks like the objects `NetworkPolicy` are not getting created,  I do not know where to put them so they can take effect : / 
```
[kni@provisioner.cluster6.dfwt5g.lab ~]$ oc -n example-cnf get NetworkPolicy
No resources found in example-cnf namespace.

[kni@provisioner.cluster6.dfwt5g.lab ~]$ oc get NetworkPolicy -A 
No resources found
```
Also maybe you are just testing but the policies would only cover the pods with  `control-plane=controller-manager` label, then based on the error I believe we'll need policies for the loadbalancer, testpmd-app, and trexconfig pods : )
```
[kni@provisioner.cluster6.dfwt5g.lab ~]$ oc -n example-cnf get pods -l control-plane=controller-manager
NAME                                                      READY   STATUS    RESTARTS      AGE
cnf-app-mac-operator-controller-manager-d76cf5457-n27n7   1/1     Running   2 (50m ago)   54m
testpmd-lb-operator-controller-manager-65786455c6-ck845   1/1     Running   0             53m
testpmd-operator-controller-manager-b95d64766-478xt       1/1     Running   0             54m
trex-operator-controller-manager-76cdd54f45-8bl8c         1/1     Running   0             53m

```

**Comment by nsilla** [comment]
Hi, Manny.

_Also maybe you are just testing but the policies would only cover the pods with control-plane=controller-manager label, then based on the error I believe we'll need policies for the loadbalancer, testpmd-app, and trexconfig pods : )_

Yes, right now I'm only testing the cnf-app-mac-operator. Once this operator passes the tests, I'll add the networkpolicies for every other operator.

**Comment by nsilla** [comment]
Something I just noticed is the tnf_test_example jobs that do pass the deny-all network policy, have a specific [hook action to create the networkpolicies](https://github.com/redhat-cip/dci-openshift-app-agent/blob/d072566ddc7cd8bc7af52372a1eec8ad913c3f03/samples/tnf_test_example/hooks/deploy_resources_in_ns.yml#L142C1-L145C105), instead of having them attached to the operators configuration.

We copied the network policies present in the TNF operator for the cnf app mac operator following Ramon's advice, but it may be the TNF operator is not properly configured either.

**Comment by ramperher** [comment]
> Something I just noticed is the tnf_test_example jobs that do pass the deny-all network policy, have a specific [hook action to create the networkpolicies](https://github.com/redhat-cip/dci-openshift-app-agent/blob/d072566ddc7cd8bc7af52372a1eec8ad913c3f03/samples/tnf_test_example/hooks/deploy_resources_in_ns.yml#L142C1-L145C105), instead of having them attached to the operators configuration.
> 
> We copied the network policies present in the TNF operator for the cnf app mac operator following Ramon's advice, but it may be the TNF operator is not properly configured either.

I told @nsilla offline, but just to note that tnf_test_example is **not** based on an operator deployment.

Also, I had issues in the deployment of some of my changes where I want to deploy extra resources. I think we need to dedicate some time to review this, since it's a problem we're having multiple people.

**Comment by nsilla** [comment]
Currently testing a new approach to set the required network policies from the app-agent's pre-run hooks in [this PR](https://github.com/dci-labs/example-cnf-config/pull/93).

**Comment by ramperher** [comment]
@nsilla , I finally figured it out how to provide these resources, you can follow the same approach I've done in https://github.com/openshift-kni/example-cnf/pull/47

Mainly, you need to create the resource in the role of each operator and allow the new resources in the controller-manager's roles files.

**Comment by nsilla** [comment]
> @nsilla , I finally figured it out how to provide these resources, you can follow the same approach I've done in #47. With this, you won't need to create the resources in the hooks.
> 
> Mainly, you need to create the resource in the role of each operator and allow the new resources in the controller-manager's roles files.

So, basically, it's the operator itself who must create the network policies? I was guessing that'll be the way to go, but it'll be a bit more difficult with the go operators, right?

**Comment by ramperher** [comment]
> > @nsilla , I finally figured it out how to provide these resources, you can follow the same approach I've done in #47. With this, you won't need to create the resources in the hooks.
> > Mainly, you need to create the resource in the role of each operator and allow the new resources in the controller-manager's roles files.
> 
> So, basically, it's the operator itself who must create the network policies? I was guessing that'll be the way to go, but it'll be a bit more difficult with the go operators, right?

It depends. In our case, where we only have one namespace and all the workloads are deployed there, maybe the simplest approach could be to directly define the network policies in the example-cnf hooks, just allowing traffic within the example-cnf namespace, or with example-cnf and openshift-dns, to allow DNS resolution (like I did in tnf_test_example with Aspenmesh: https://github.com/redhat-cip/dci-openshift-app-agent/blob/master/samples/tnf_test_example/hooks/templates/network_policies_am.yml.j2), and reject the traffic with other namespaces.

**Comment by ramperher** [comment]
BTW plese rebase the PR to avoid issues with this change already merged: https://github.com/openshift-kni/example-cnf/pull/62

File: testpmd-operator/roles/testpmd/tasks/main.yml [hunk 1]
```diff
@@ -79,6 +79,11 @@
     state: present
     definition: "{{ lookup('template', 'role_binding.yml') }}"
 
+- name: Create default network policies
+  community.kubernetes.k8s:
```
Comment (ramperher) [inline testpmd-operator/roles/testpmd/tasks/main.yml:83]: you need to use `kubernetes.core.k8s`, that's why it's failing

**Review by ramperher — CHANGES_REQUESTED** [review]
@nsilla , in your last tests, the testpmd-operator-controller-manager is not even able to be deployed, it remains in CrashLoopBackOff status.
Maybe, it could be a good exercise to start testing the creation of the NetworkPolicies you're defining in your change from https://github.com/dci-labs/example-cnf-config/pull/93, verifying they're working fine, and then, you can really move them here if required.
Remember that, right now, you're only covering the controller-manager pods. You also need to create NetworkPolicies for all the other pods in the scenario. These are the labels you have to have in mind:
- example-cnf-type=lb-app - from testpmd-lb-operator
- example-cnf-type=cnf-app - from testpmd-operator
- example-cnf-type=pkt-gen - from trex-operator
- example-cnf-type=pkt-gen-app - from trex-operator

**Comment by manurodriguez** [comment]
closing this, integration don in dci-labs/example-cnf-config#93

**Pull request closed without merging**